contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,293 @@
+		logging.GetLogger().Errorf("vpclogs interfaces not well defined")
+		return nil, common.ErrBadArgument
+	}
+
+	// get TIDs of specified interfaces
+	tid1, err := GetInterfaceTid(accRejClient, host, interfaces[0])
+	if err != nil {
+		return nil, err
+	}
+	tid2, err := GetInterfaceTid(accRejClient, host, interfaces[1])
+	if err != nil {
+		return nil, err
+	}
+	logging.GetLogger().Infof("NewAccRejAction, tid1 = %s", tid1)
+	logging.GetLogger().Infof("NewAccRejAction, tid2 = %s", tid2)
+
+	// start a capture on the specified interfaces
+	captureId, err := StartCapture(accRejClient, tid1, tid2)
+	if err != nil {
+		return nil, err
+	}
+	logging.GetLogger().Infof("NewAccRejAction, captureId = %s", captureId)
+	// TODO need to pass the captureId to the websocket interface
+
+	// allocate the table to store flow state
+	accRejState := make(map[string]*accRejFlowInfo)
+	accRej := &accRejAction{
+		accRejClient: accRejClient,
+		accRejState:  accRejState,
+		captureId:    captureId,
+	}
+	return accRej, nil
+}
+
+// InitAccRejAction - initialize internal data structure for this round
+func (t accRejAction) InitAccRejAction() {
+	// reset the entries in the table for this round
+	for _, ars := range t.accRejState {
+		ars.Saw1 = false
+		ars.Saw2 = false
+		ars.Status = FlowInfo_OLD_FLOW
+	}
+}
+
+// AccumulateAccRejInfo - go over all the flows and fill in our internal data structure with relevant info
+func (t accRejAction) AccumulateAccRejInfo(fl []*flow.Flow) {
+	for _, f := range fl {
+		L3TrackingID := f.GetL3TrackingID()
+		flowInf, ok := t.accRejState[L3TrackingID]
+		if !ok {
+			// this flow is not yet in our table
+			t.accRejState[L3TrackingID] = &accRejFlowInfo{
+				UUID1:   f.UUID,
+				Metric1: *f.Metric,
+				Status:  FlowInfo_ONE_ENTRY,
+				Saw1:    true,
+				Flow1:   f,
+			}
+			continue
+		} else if flowInf.Status == FlowInfo_ONE_ENTRY {
+			if flowInf.UUID1 != f.UUID {
+				flowInf.UUID2 = f.UUID
+				flowInf.Metric2 = *f.Metric
+				flowInf.Saw2 = true
+				flowInf.Flow2 = f
+				flowInf.Status = FlowInfo_TWO_ENTRIES
+			} else {
+				// got duplicate from same flow; need to indicate reject
+				// or accumulate the stats until next time
+				flowInf.Metric1 = *f.Metric
+				flowInf.Saw1 = true
+				flowInf.Flow1 = f
+				continue
+			}
+		} else { // normal case
+			if flowInf.UUID1 == f.UUID {
+				flowInf.Metric1 = *f.Metric
+				flowInf.Flow1 = f
+				flowInf.Saw1 = true
+			} else {
+				flowInf.Metric2 = *f.Metric
+				flowInf.Flow2 = f
+				flowInf.Saw2 = true
+			}
+		}
+	}
+}
+
+// combineFlows - helper function to combine 2 seen flows into one, based on common minimums
+func combineFlows(f *accRejFlowInfo) *flow.Flow {
+	// perform shallow copy of flow and then update some fields
+	newF := *f.Flow1
+	newMetricReported := *f.Flow1.Metric
+	// take byte and packet count as minimum of the 2 reports
+	if f.Flow1.Metric.ABPackets > f.Flow2.Metric.ABPackets {
+		newMetricReported.ABPackets = f.Flow2.Metric.ABPackets
```

Comment:
Maybe a `min(a, b int64)` function can make this part clearer?

Reply:
OK